File: FLAME_project/data/compiler_technology/buggy/1/1337/correct/parser/expression/unaryexp/Number.cpp
/**
 *
 */

#include "Number.h"
#include "../../../token/Token.h"
#include "../../../util/Util.h"
#include <fstream>

void Number::numberParser() {
    tokenItem token = Token::getNowToken();
    if(token.type == "INTCON"){
       token = Token::getNextToken();
    }
    Util::printParserInfo("<Number>");
}

File: FLAME_project/data/compiler_technology/buggy/1/1350/correct/nonTerminal/Number.java
package nonTerminal;

import frontend.Lexer;
import frontend.Token;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;

public class Number {
    private Token intcon;
    private int lineCnt;

    public Number() {
        this.intcon = null;
        this.lineCnt = 0;
    }

    public void parser(Lexer lexer) {
        intcon = lexer.now();
        lexer.move();
        lineCnt = intcon.getLineCnt();
    }

    public void output(ArrayList<String> parserResult) {
        parserResult.add(intcon.getType().toString() + " " + intcon.getContent());
        parserResult.add("<Number>");
    }

    public int getLineCnt() {
        return lineCnt;
    }
}
